src/messages.lisp

```diff
@@ -88,6 +88,20 @@
      )
   :documentation "A single request object according to the JSONRPC standard.")
 
+(defmessage |RPCWarning| ()
```

Comment:
errors in rpcq just have a string, whereas warnings also have a kind. What are the possible kinds for warnings and why don't errors need this same functionality?

Reply:
Right now, I'm populating `kind` with the stringified name of the signal class being raised (which in practice tends to be more expressive on the Lisp side than on the python side). I'd be happy for errors to also have similarly populated `kind` slots.

Reply:
Well so errors are fully serialized in some format that allows getting the error name, message, and traceback.

I think I'm happy to keep this asymmetry given that warnings won't have a stack trace. BUT I'm still skeptical of the value/utility of "kind" - given that it will almost always be `UserWarning` when people call warnings.warn without a type

Reply:
I feel very lukewarm about this view of error serialization. We're writing a serialization library, so should we try to make use of it when serializing things instead of packing all that info into a string? The Lisp server doesn't make any effort to conform to an existing standard of name+message+traceback packing, in large part bc I thought that this unstructured string discouraged anyone from thinking they could rely on its format. On the other hand, Lisp does strongly encourage the use of custom error/warning classes, and I'd like to capture that information somewhere.

Reply:
So obviously the way we do error serialization is bad - but it would be a good amount of effort to change this now and be backwards compatible. Worth filing an issue though.

Fair enough, we can keep kind in for now. Then in a perfect future error would have kind as well and a separate structured array for tracebacks.

Reply:
Opened as #46 .